test/jdk/java/awt/Robot/MouseMoveOffScreen.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.GraphicsConfiguration;
+import java.awt.GraphicsDevice;
+import java.awt.GraphicsEnvironment;
+import java.awt.MouseInfo;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+
+/*
+ * @test
+ * @bug 8345538
+ * @summary Tests mouseMove clamping to screen bounds when set to move offscreen
+ * @key headful
+ * @run main MouseMoveOffScreen
+ */
+
+public class MouseMoveOffScreen {
+    private static final Point STARTING_LOC = new Point(200, 200);
+    private static final Point OFF_SCREEN_LOC = new Point(20000, 200);
+    private static Rectangle[] r;
+
+    public static void main(String[] args) throws Exception {
+        GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
+        GraphicsDevice[] gs = ge.getScreenDevices();
+        r = new Rectangle[gs.length];
+
+        for (int i = 0; i < gs.length; i++) {
+            r[i] = gs[i].getDefaultConfiguration().getBounds();
+        }
```

Comment:
I would suggest logging screen bounds (provides better debug info for anyone running the test)

```suggestion
        for (int i = 0; i < gs.length; i++) {
            r[i] = gs[i].getDefaultConfiguration().getBounds();
            System.out.println("Screen: "+ gs[i].getIDstring() + "  Bounds: " + r[i]);
        }
```